src/plugin.cpp

```diff
@@ -0,0 +1,29 @@
+#include "./plugin.hpp"
+#include "auth_plugin.h"
+#include "bucket_plugin.h"
+
+#include <dlfcn.h>
+#include <iostream>
+#include <fmt/format.h>
+
+namespace irods::s3::plugins
+{
+    void load_plugin(rcComm_t& connection, const std::string_view& plugin_name, nlohmann::json& configuration)
+    {
+        auto i = fmt::format("lib{}.so", plugin_name);
+        void* plugin_handle = dlopen(i.c_str(), RTLD_LAZY | RTLD_LOCAL);
+        if (plugin_handle == nullptr) {
+            std::cerr << "Plugin library " << std::quoted(i) << " was not found by the linker" << std::endl;
+        }
+        void (*init)(rcComm_t*, const char*);
+
+        // I love C :3
+        *(void**) (&init) = dlsym(plugin_handle, "plugin_initialize");
+        if (init == nullptr) {
+            std::cerr << "Plugin library " << std::quoted(i)
+                      << " does not contain an exported 'plugin_initialize' symbol." << std::endl;
+        }
+        auto dump = configuration.dump();
+        (*init)(&connection, dump.c_str());
```

Comment:
Why not use Boost.DLL for loading shared libraries?

Reply:
Because I didn't need to read docs to know how to hold dlopen due to past experience.

Would you prefer if I did it through Boost DLL? I've not really compared what I get with that versus dlopen so I'm absolutely not opposed to switching over to that.

Reply:
I was just wondering. I think it's worth a look, but it's not the highest priority item for this work.

Let's capture the investigation of Boost.DLL in a new issue and handle it later. No need to get sidetracked by shared library loading.